### PR TITLE
Honor "secrets until the end" check in frontend

### LINKED ISSUE #59: Honor "secrets until the end" check in frontend
Ref: VOC-SEC-07

Since 95ca1e1e79036c1763dac22463614cd4495c8563 we have the setting `secretUntilTheEnd`. This means that the results will be published in real time through the Explorer Vote portal.

For making this more transparent to all the voters, these results should be accessible from the Decidim installation.

### Acceptance criteria

- [ ] Given that I'm a voter,
      When I go to an open Election with this setting not enable,
      Then I can see the results

### PR DESCRIPTION
**Pull request by antopalidi**
fixes #59 

Screenshot 2023-07-18 at 15 26 44

Screenshot 2023-07-19 at 15 33 48

### CODE REVIEW

**Comment by antopalidi** [comment]
@andreslucena, this pr is ready for review

File: app/views/decidim/vocdoni/admin/election_calendar/edit.html.erb [hunk 1]
```diff
@@ -8,26 +8,37 @@
...
-        <div class="row column">
+        <div class="row column mb-s">
           <%= f.datetime_field :end_time %>
         </div>
-        <div class="row column">
-          <%= f.check_box :secret_until_the_end, { checked: election.election_type&.fetch("secret_until_the_end", false) } %>
-        </div>
         <div class="row column">
           <%= f.check_box :interruptible, { checked: election.election_type&.fetch("interruptible", true) } %>
         </div>
```
[the remark this answers is not in the document]
Comment (carolromero) [reply app/views/decidim/vocdoni/admin/election_calendar/edit.html.erb:30]: @andreslucena @antopalidi I confirm it!

File: app/forms/decidim/vocdoni/admin/election_calendar_form.rb
[the remark this answers is not in the document]
Comment (antopalidi) [reply app/forms/decidim/vocdoni/admin/election_calendar_form.rb:1]: It's already been added there:

`secret_until_the_end: [true, false].sample`

File: app/services/decidim/vocdoni/vocdoni_client.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      def fetch_from_api(path)
+        url = "#{base_url}#{path}"
+        response = Faraday.get(url)
+
+        JSON.parse(response.body) if response.success? && response.body.present?
+      end
+
+      def base_url
+        Decidim::Vocdoni.api_endpoint_url(api_endpoint_env)
```
Comment (andreslucena) [inline app/services/decidim/vocdoni/vocdoni_client.rb:27]: Related to my other commend, we should take the env from the one already defined in Decidim::Vocdoni
```suggestion
        Decidim::Vocdoni.api_endpoint_url
```

File: app/services/decidim/vocdoni/vocdoni_client.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      attr_reader :wallet, :api_endpoint_env
+
+      def initialize(wallet:, api_endpoint_env:)
+        @wallet = wallet
+        @api_endpoint_env = api_endpoint_env
+      end
+
+      def fetch_election(vocdoni_election_id)
+        fetch_from_api("/elections/#{vocdoni_election_id}")
+      end
```
Comment (andreslucena) [inline app/services/decidim/vocdoni/vocdoni_client.rb:15]: `:wallet` isn't used, and `:api_endpoint_env`, as I mentioned in other comments, can be inferred from `Decidim::Vocdoni` already.  Also I think it's better to just send the `:vocdoni_election_id` in the initializer. This is what I have in mind:

```suggestion
      attr_reader :vocdoni_election_id

      def initialize(vocdoni_election_id:)
        @vocdoni_election_id = vocdoni_election_id
      end

      def fetch_election
        fetch_from_api("/elections/#{vocdoni_election_id}")
      end
```

File: app/controllers/decidim/vocdoni/elections_controller.rb [hunk 4]
```diff
@@ -28,6 +37,32 @@ def election
...
+      def election_data
+        @election_data ||= vocdoni_client.fetch_election(vocdoni_election_id)
+      end
+
+      def election_metadata
+        return nil unless !election.election_type["secret_until_the_end"] && election.ongoing?
+
+        @election_metadata ||= election_data
+      end
+
```
Comment (andreslucena) [inline app/controllers/decidim/vocdoni/elections_controller.rb:65]: Related to my other comments, I think we can simplify this a lot: 

```suggestion
      def vocdoni_client
        @vocdoni_client ||= VocdoniClient.new(vocdoni_election_id: election.vocdoni_election_id)
      end

      def election_metadata
        return nil unless !election.election_type["secret_until_the_end"] && election.ongoing?

        @election_metadata ||= vocdoni_client.fetch_election
      end
```

**Review by andreslucena — CHANGES_REQUESTED** [review]
I have some more suggestions and I think this is ready 😄

**Comment by antopalidi** [comment]
@andreslucena all done

**Review by andreslucena — APPROVED** [review]
LGTM!! 
@microstudi can you take a look 🙏🏽 ?

**Review by microstudi — COMMENTED** [review]
Hi, the secret until the end feature looks great!

I have a concern, I see that the value "interruptible" comes from the module settings now and it is not configurable from the component.

Is there some reason for this change? maybe i am missing something.

**Comment by andreslucena** [comment]
> Is there some reason for this change? maybe i am missing something.

As we've discussed in the Telegram chat, we've talked about not giving Admins this feature as it's something that they'd want to use almost always. 

If an installation needs to change this behaviour, they can do so by changing this in the initializer, so that works for that use case.

**Pull request merged**